worker/docker_image_manager.py

```diff
@@ -0,0 +1,135 @@
+import threading
+import os
+import time
+import traceback
+import logging
+import json
+
+import file_util
+
+logger = logging.getLogger(__name__)
+
+
+class DockerImageManager(object):
+    """
+    Manages the Docker images available on the worker, and ensures that they
+    do not fill up the disk.
+
+    Warning: Assumes that Docker images are stored on the same disk as the
+    work directory specified by the `workdir` parameter. If this is not the
```

Comment:
That doesn't work on Mac -- but like I mentioned in the issue thread I have no qualms about forgoing Mac support for this feature.